core/controllers/suggestion_test.py

```diff
@@ -992,6 +994,375 @@ def test_translation_suggestion_creation_with_new_images(self) -> None:
+                '</oppia-noninteractive-image>')
+        }
+        self.login(self.EDITOR_EMAIL)
+        csrf_token = self.get_new_csrf_token()
+
+        with utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'),
+            'rb', encoding=None
+        ) as f:
+            raw_image = f.read()
+        self.post_json(
+            '%s/exploration/%s' % (
+                feconf.EXPLORATION_IMAGE_UPLOAD_PREFIX, exp_id),
+            {'filename': 'img.png'},
+            csrf_token=csrf_token,
+            upload_files=[('image', 'unused_filename', raw_image)])
+        exp_services.update_exploration(
+            self.editor_id, exp_id, [exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                'property_name': exp_domain.STATE_PROPERTY_CONTENT,
+                'state_name': 'State 1',
+                'new_value': state_content_dict
+            })], 'Changes content.')
+        rights_manager.publish_exploration(self.editor, exp_id)
+
+        exploration = exp_fetchers.get_exploration_by_id(exp_id)
+        text_to_translate = exploration.states['State 1'].content.html
+        self.logout()
+
+        valid_html = (
+            '<oppia-noninteractive-math math_content-with-value="{&amp;q'
+            'uot;raw_latex&amp;quot;: &amp;quot;(x - a_1)(x - a_2)(x - a'
+            '_3)...(x - a_n-1)(x - a_n)&amp;quot;, &amp;quot;svg_filenam'
+            'e&amp;quot;: &amp;quot;file.svg&amp;quot;}"></oppia-noninte'
+            'ractive-math>'
+        )
+        large_image = '<svg><path d="%s" /></svg>' % (
+             'M150 0 L75 200 L225 200 Z ' * 4000)
+        self.login(self.TRANSLATOR_EMAIL)
+        csrf_token = self.get_new_csrf_token()
+        response_dict = self.post_json(
+            '%s/' % feconf.SUGGESTION_URL_PREFIX, {
+                'suggestion_type': (
+                    feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT),
+                'target_type': feconf.ENTITY_TYPE_EXPLORATION,
+                'target_id': exp_id,
+                'target_version_at_submission': exploration.version,
+                'change_cmd': {
+                    'cmd': exp_domain.CMD_ADD_WRITTEN_TRANSLATION,
+                    'state_name': 'State 1',
+                    'content_id': 'content_0',
+                    'language_code': 'hi',
+                    'content_html': text_to_translate,
+                    'translation_html': valid_html,
+                    'data_format': 'html'
+                },
+                'description': 'test',
+                'files': {'file.svg': large_image},
+            }, csrf_token=csrf_token, expected_status_int=400,
+        )
+
+        self.assertIn(
+            'Image exceeds file size limit of 100 KB.',
+            response_dict['error'])
+        self.logout()
+
+    def test_saves_new_images_in_translation_that_were_copied_from_target_exploration( # pylint: disable=line-too-long
```

Comment:
Try adding a test where the source image is replaced by a different image in the suggestion (or where there are two images in the original text and only one is replaced and the other is kept).

Reply:
@seanlip Doesn't the test named `test_saves_new_images_in_translation_that_were_uploaded_by_submitter` already cover such a case?

Reply:
@StephenYu2018 Sorry, I can't find that test ... which line of code are you referring to?

Reply:
@seanlip Line 860 within `core/controllers/suggestion_test.py` on my issue branch (it's not on `develop` branch since I renamed the test).

Reply:
Also, are you referring to the exploration editor changing the image in the exploration when you say the source image is replaced (like what you clarified in a different review thread)? If so, then my initial comment of this thread doesn't apply and I'll have to come up with a new test for this case.

Reply:
Yup, that is correct -- that's what I'm referring to. Thanks for checking!

Reply:
@seanlip Since you're allowing me to open a separate issue to fix the translation suggestion image pipeline as stated in [this comment](https://github.com/oppia/oppia/pull/20188#discussion_r1582058592), I think it's best to postpone writing this test case for that separate issue as opposed to this PR's linked issue. Is this review thread considered as resolved now?

Reply:
Yup, seems fine. Thanks!